openupgradelib/openupgrade_merge_records.py

```diff
@@ -431,6 +431,111 @@ def _adjust_merged_values_orm(env, model_name, record_ids, target_record_id,
+        return
+    env.cr.execute(
+        """
+        SELECT isc.column_name, isc.data_type, imf.ttype
+        FROM information_schema.columns isc
+        JOIN ir_model_fields imf ON (
+            imf.name = isc.column_name AND imf.model = %s)
+        WHERE table_name = %s
+        """, (model_table, model_name))
+    vals = {}
+    new_vals = {}
+    for column, column_type, field_type in env.cr.fetchall():
+        if (model_table, column) in exclude_columns:
+            continue
+        op = field_spec.get(column, False)
+        env.cr.execute(
+            sql.SQL(
+                """SELECT {column}
+                FROM {table}
+                WHERE id IN %(record_ids)s"""
+            ).format(
+                table=sql.Identifier(model_table),
+                column=sql.Identifier(column),
+            ), {'record_ids': (target_record_id,) + tuple(record_ids)}
+        )
+        _list = [x[0] for x in env.cr.fetchall()]
+        if field_type in ('char', 'text', 'html'):
+            if not op:
+                op = 'other' if field_type == 'char' else 'merge'
+            if op == 'merge':
+                _list = filter(lambda x: x is not False, _list)
+                vals[column] = ' | '.join(_list)
+        elif field_type in ('integer', 'float', 'monetary'):
+            if not op:
+                op = 'other' if field_type == 'integer' else 'sum'
+            if op == 'sum':
+                vals[column] = sum(_list)
+            elif op == 'avg':
+                vals[column] = sum(_list) / len(_list)
+            elif op == 'max':
+                vals[column] = max(_list)
+            elif op == 'min':
+                vals[column] = min(_list)
+        elif field_type == 'boolean':
+            op = op or 'other'
+            if op == 'and':
+                vals[column] = functools.reduce(lambda x, y: x & y, _list)
+            elif op == 'or':
+                vals[column] = functools.reduce(lambda x, y: x | y, _list)
+        elif field_type in ('date', 'datetime'):
+            if op:
+                _list = filter(lambda x: x is not False, _list)
+            op = _list and op or 'other'
+            if op == 'max':
+                vals[column] = max(_list)
+            elif op == 'min':
+                vals[column] = min(_list)
+        elif field_type == 'binary':
+            op = op or 'merge'
+            if op == 'merge':
+                first_value = _list[0]
+                _list = [x for x in _list if x]
+                if not first_value and _list:
+                    vals[column] = _list[0]
+        elif field_type == 'many2one':
+            op = op or 'merge'
+            if op == 'merge':
+                first_value = _list[0]
+                if _list and not first_value:
+                    vals[column] = _list[0]
+        # Curate values that haven't changed
+        if vals[column] != _list[0]:
+            new_vals[column] = vals[column]
+    if new_vals:
+        logged_query(
```

Comment:
Not sure if this query should be logged, and if so, then don't double log after in debug mode. Isn't the exactly same detail?